.github/workflows/test.yaml

```diff
@@ -6,32 +6,27 @@ on:
       - main
 
 jobs:
-  test-publish-hex:
-    name: It publishes a hex package in a folder
+  test-publish-cpan:
+    name: Publishes a perl module to CPAN from a dist.ini
     runs-on: ubuntu-latest
-    environment: Publish to hex
+    environment: Publish to CPAN
     steps:
       - uses: actions/checkout@v2
-      - uses: erlef/setup-beam@v1
-        with:
-          otp-version: 25.0.3
-          elixir-version: 1.13.4
-      - run: mix deps.get
-        working-directory: "elixir"
       - name: Set unique package version
         id: set-version
         run: |
           version=0.0.$GITHUB_RUN_NUMBER
-          mix bump $version
+          sed -i "s/0.0.0/$version/g" VERSION
           echo "::set-output name=version::$version"
-        working-directory: "elixir"
+        working-directory: "perl"
       - name: Test the action
         uses: ./
         with:
-          hex-api-key: ${{ secrets.HEX_API_KEY }}
-          working-directory: "elixir"
-      - name: Assert that latest published package version is as expected
+          cpan-user: ${{ secrets.CPAN_USER }}
+          cpan-password: ${{ secrets.CPAN_PASSWORD }}
+          working-directory: "perl"
+      - name: Assert that latest published package version is available on CPAN
         timeout-minutes: 1
         run: |
           version=0.0.$GITHUB_RUN_NUMBER
-          curl --silent --fail https://hex.pm/packages/cucumber_test_release_automation/$version > /dev/null
+          curl --silent --fail https://metacpan.org/release/CUKEBOT/Cucumber-Test-Release-Automation-v$version > /dev/null
```

Comment:
I'm actually expecting this to work only after a certain amount of time:  releases go through an indexing process before becoming publicly visible. This period is usually really short, but the system lags from time to time, which means that the check may not work. It should normally work if you do this check after 15 to 20 minutes instead of immediately.

Reply:
Ooh gosh that's a long time. I'm trying it with am eventually loop now (1 minute timeout).

Reply:
Yeah a one-minute timeout didn't work, I'll try 10 minutes.

Reply:
Note that we only need to wait for this delay when we make changes to this github action, which should hopefully happen very rarely if at all.

Reply:
Wow so the last one that I can see right now is https://github.com/cucumber/action-publish-cpan/runs/7867037793?check_suite_focus=true#step:5:115 which was pushed 33 minutes ago. None of the subsequent ones are visible yet.

Reply:
OK so 10 minutes didn't work, I'm now trying 60.

@ehuelsmann can you think of another way we could do a functional test of the action?

Reply:
How about checking https://www.cpan.org/modules/by-authors/id/C/CU/CUKEBOT/ ? 

That should update much quicker.

Reply:
Much quicker being "well before the cpan index is updated".